### ISSUE TITLE
[typedef] Add support for call-signature

### ISSUE THREAD

**Issue by Robouste** [typescript-eslint/typescript-eslint#3650]
- [ X] I have tried restarting my IDE and the issue persists.
- [ X] I have updated to the latest version of the packages.
- [X ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
        "typedef": [true, "call-signature"],
        "@typescript-eslint/typedef": [
	        "warn",
	        {
		        "call-signature": true
	        }
        ],
  }
}
```

```TS
export class Testing {
	public someMethod() {
		
	}
}
```

**Expected Result**

I should get a warning telling me I haven't specified the return type of the `someMethod()` method.

**Actual Result**

Nothing happens

**Additional Info**

I have asked the question on stackoverflow:
[Thread](https://stackoverflow.com/questions/68374703/typescript-eslint-typedef-enable-call-signature)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.28.4` |
| `@typescript-eslint/parser`        | `4.28.4` |
| `TypeScript`                       | `4.2.3` |
| `ESLint`                           | `7.31.0` |
| `node`                             | `12.19.0` |

**Comment by bradzacher** [comment]
Have you read our docs at all?
We have a different lint rule for doing exactly this.
`explicit-function-return-type`
